### PR TITLE
fixed some typo mistakes in the architecture guide

### PR DESCRIPTION
**Pull request by ibilalkayy**
### What's being changed:

In this Pull Request, I fixed some of the typo mistakes in the architecture guide. The sentences were not making sense due to the typo mistakes. So I fixed them.

### Type of change:

- [x] **Documentation** updates (non-breaking change to fix/update documentation)
- [ ] **Website** updates (non-breaking change to update main page, company pages, pricing, etc)
- [ ] **Content** updates – **blog**, **podcast** (non-breaking change to add/update content)
- [ ] **Bug fix** (non-breaking change to fixes an issue with the site)
- [ ] **Feature** or **enhancements** (non-breaking change to add functionality)

### How Has This Been Tested?

- [ ] **Github action** – automated build completed without errors
- [x] **Local build** - the site works as expected when running `yarn start`

> note, you can run `yarn verify-links` to test site links locally

### CODE REVIEW

File: developers/contributor-guide/weaviate-modules/architecture.md [hunk 1]
```diff
@@ -59,22 +59,21 @@ capability interface methods.
...
 * The struct `AdditionalProperty` defines all the needed settings and methods to be set in order to add the additional property to Weaviate.
-  * `RestNames              []string`: look `handlers_kinds.go` - defines rest api parameter names
-	* `DefaultValue           interface{}`: look `handlers_kinds.go` - defines a default value for a parameter
-	* `GraphQLNames           []string`: graphql additional parameter name
-	* `GraphQLFieldFunction   GraphQLFieldFn`: defines the additional property graphql argument
-	* `GraphQLExtractFunction ExtractAdditionalFn`: defines the extract function for additional property argument
-	* `SearchFunctions        AdditionalSearch`: defines all of the functions for rest api and graphql
-* The interface `AdditionalProperties` groups whole interface methods needed for adding the capability of additional properties, and defines all of the additional property parameters.
-
+  * `RestNames              []string`: look `handlers_kinds.go` - defines rest api parameter names.
```
Comment (daveatweaviate) [inline developers/contributor-guide/weaviate-modules/architecture.md:70]: Pre-existing issue: please delete the '-' before defines.

File: developers/contributor-guide/weaviate-modules/architecture.md [hunk 2]
```diff
@@ -59,22 +59,21 @@ capability interface methods.
...
+	* `DefaultValue           interface{}`: look `handlers_kinds.go` - defines a default value for a parameter.
```
Comment (daveatweaviate) [inline developers/contributor-guide/weaviate-modules/architecture.md:71]: Pre-existing issue: please delete the '-' before defines.

**Review by daveatweaviate — COMMENTED** [review]
@ibilalkayy thank you for your suggestions. I left comments about two small, pre-existing issues in the updated lines. Would you mind fixing them with your PR?

Would you accept the a [Contributor License Agreement](https://weaviate.io/service/contributor-license-agreement) in this PR?

I know you agreed to it in #1504 but we need an acknowledgement for each PR.  

Thanks!

**Comment by ibilalkayy** [comment]
I agree with the CLA.

**Comment by daveatweaviate** [comment]
@ibilalkayy thanks for your updates, I've merged these in!

**Pull request merged**